Optimizely test JVM hangs on exit (leaked non-daemon threads) — true cause of dangling CI runs

## Symptom

Intermittently, a `build-matrix` cell runs all tests to completion but the test JVM never exits, so the job hangs until the job timeout (now 20 min, #216) kills it. Before #216 this was a ~6h dangling run.

Evidence — run [27474648449](https://github.com/EtaCassiopeia/zio-openfeature/actions/runs/27474648449), cell `build-matrix (17, 2.13.16)`:

```
18:02:45  + Optimizely datafile polling lifecycle (#208)     <- last test, all passed
18:21:01  ##[error]The operation was canceled.               <- 20-min timeout
18:21:01  Terminate orphan process: pid (2684) (java)        <- JVM never exited
```

All tests passed; the JVM then idled ~18 minutes until killed. `Terminate orphan process: (java)` confirms a live non-daemon thread kept it alive.

## Root cause

The `optimizely` test suite leaves non-daemon threads running after tests finish — almost certainly the Optimizely `HttpProjectConfigManager`'s datafile-polling `ScheduledExecutorService` (and/or a WireMock Jetty pool) from a provider/server that wasn't fully closed on some path. sbt runs tests in-process by default, so a leaked non-daemon thread blocks JVM exit and hangs the whole `test` task. Intermittent because it depends on scheduling/whether the poller had spun a thread.

This is distinct from #211 (the flaky `requests == 0` assertion) — that was a fast failure; this is a hang.

## Suggested fix (pick one)

1. **`optimizely / Test / fork := true`** — forked test JVMs are force-terminated by sbt after the run, so leaked threads can't hang the build. Smallest, most robust. (Consider the same for `ofrep`/`extras`, which also spin WireMock.)
2. Ensure every constructed Optimizely client / config manager is closed (daemonize the poller threads, or guarantee `shutdown()`/scope close on all test paths).

Option 1 is the reliable safety net; option 2 addresses the leak at the source. Ideally both.

--- a/build.sbt
+++ b/build.sbt
@@ -90,6 +90,12 @@ lazy val crossVersionSourceDirs = Seq(
 
 lazy val commonSettings = Seq(
   testFrameworks += new TestFramework("zio.test.sbt.ZTestFramework"),
+  // Run tests in a forked JVM. A forked test run is a child process that sbt force-terminates when the `test` task
+  // completes, so a leaked non-daemon thread (a poller, a WireMock/Jetty pool, anything) can't block exit the way it
+  // can in-process. This is the categorical fix for #217 and #229: both issues are repeated instances of "some
+  // non-daemon thread outlived the test run and hung the JVM," and chasing each individual leak source (already done
+  // across several merged PRs) only closes the specific case found, not the class of bug.
+  Test / fork := true,
   libraryDependencies ++= Seq(
     "dev.zio" %% "zio-test"     % zioVersion % Test,
     "dev.zio" %% "zio-test-sbt" % zioVersion % Test

